packages/react-conformance/src/defaultTests.tsx

```diff
@@ -327,6 +327,129 @@ export const defaultTests: TestObject = {
+        // Need to work on this error message
+        throw new Error(
+          defaultErrorMessages['component-has-static-classnames-object-exported'](
+            testInfo,
+            e,
+            componentClassName,
+            exportName,
+          ),
+        );
+      }
+    });
+
+    it('has static classnames in correct format (component-has-static-classnames-object)', () => {
+      if (!handledClassNamesObjectExport) {
+        return;
+      }
+
+      const exportName =
+        componentInfo.displayName.slice(0, 1).toLowerCase() + componentInfo.displayName.slice(1) + 'ClassNames';
+      const indexFile = require(path.join(getPackagePath(componentPath), 'src', 'index'));
+      const classNamesFromFile = indexFile[exportName];
+
+      const expectedClassNames = Object.keys(classNamesFromFile).reduce(
+        (obj: { [key: string]: string }, key: string) => {
+          obj[key] = key === 'root' ? componentClassName : `${componentClassName}__${key}`;
+          return obj;
+        },
+        {},
+      );
+
+      try {
+        expect(classNamesFromFile).toEqual(expectedClassNames);
+      } catch (e) {
+        // Need to work on this error message
+        throw new Error(
+          defaultErrorMessages['component-has-static-classnames-in-correct-format'](
+            testInfo,
+            e,
+            componentClassName,
+            exportName,
+          ),
+        );
+      }
+    });
+
+    it(`has static classnames in rendered component (component-has-static-classnames-object)`, () => {
+      if (!handledClassNamesObjectExport) {
+        return;
+      }
+
+      const { testOptions = {} } = testInfo;
+      const staticClassNameVariants = testOptions['has-static-classnames'] ?? [{ props: {} }];
+
+      for (const staticClassNames of staticClassNameVariants) {
+        const mergedProps = {
+          ...requiredProps,
+          ...staticClassNames.props,
+        };
+        const defaultEl = customMount(<Component {...mergedProps} />);
+        const defaultComponent = getComponent(defaultEl, helperComponents, wrapperComponent);
+
+        const exportName =
+          componentInfo.displayName.slice(0, 1).toLowerCase() + componentInfo.displayName.slice(1) + 'ClassNames';
+        const indexFile = require(path.join(getPackagePath(componentPath), 'src', 'index'));
+        const classNamesFromFile = indexFile[exportName];
+
+        const missingClassNames = [];
+        let className;
+        let err: Error;
+
+        const expectedClassNames = staticClassNames.expectedClassNames ?? classNamesFromFile;
+        for (const key of Object.keys(expectedClassNames)) {
+          className = classNamesFromFile[key];
+          try {
+            expect(defaultComponent.find(`.${className}`).length).toBeGreaterThanOrEqual(1);
+          } catch (e) {
+            err = e as Error;
+            missingClassNames.push(className);
+          }
+        }
+
+        if (missingClassNames.length > 0) {
```

Comment:
Above all the missing class names are collected in a single test run to avoid the situation where you fix one error in a test case and immediately get another error on the next run. My feeling is that this makes for a nicer DX but I feel that the test is harder to read as a result. Definitely open to ideas here.

Reply:
There's no real reason to use `expect` and then try/catch.

You could just do smth like
```ts
const missingClasssNames = Object.entries(expectedClassNames).reduce((acc, [key, className]) => {
  if (defaultComponent.find(className).length !== 1) {
    acc.push(className);
  }

  return acc;
}, [])
```